plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/TestCachingJdbcClient.java

```diff
@@ -309,6 +317,108 @@ public void testColumnsNotCachedWhenCacheDisabled()
+        ConnectorSession session = createSession("first");
+
+        JdbcTableHandle first = createTable(new SchemaTableName(schema, "first"));
+        JdbcTableHandle second = createTable(new SchemaTableName(schema, "second"));
+
+        int expectedLoad = 0, expectedHit = 0, expectedMiss = 0;
+        assertCacheLoadsHitsAndMisses(cachingJdbcClient.getStatisticsCacheStats(), expectedLoad, expectedHit, expectedMiss);
+
+        // load first
+        assertThat(cachingJdbcClient.getTableStatistics(session, first, TupleDomain.all())).isEqualTo(NON_EMPTY_STATS);
+        assertCacheLoadsHitsAndMisses(cachingJdbcClient.getStatisticsCacheStats(), expectedLoad, expectedHit, expectedMiss += 1);
+
+        // read first from cache
+        assertThat(cachingJdbcClient.getTableStatistics(session, first, TupleDomain.all())).isEqualTo(NON_EMPTY_STATS);
+        assertCacheLoadsHitsAndMisses(cachingJdbcClient.getStatisticsCacheStats(), expectedLoad, expectedHit += 1, expectedMiss);
+
+        // load second
+        assertThat(cachingJdbcClient.getTableStatistics(session, second, TupleDomain.all())).isEqualTo(NON_EMPTY_STATS);
+        assertCacheLoadsHitsAndMisses(cachingJdbcClient.getStatisticsCacheStats(), expectedLoad, expectedHit, expectedMiss += 1);
+
+        // invalidate first
+        cachingJdbcClient.dropTable(SESSION, first);
+        first = createTable(new SchemaTableName(schema, "first"));
+
+        // load first again
+        assertThat(cachingJdbcClient.getTableStatistics(session, first, TupleDomain.all())).isEqualTo(NON_EMPTY_STATS);
+        assertCacheLoadsHitsAndMisses(cachingJdbcClient.getStatisticsCacheStats(), expectedLoad, expectedHit, expectedMiss += 1);
+
+        // cleanup
+        this.jdbcClient.dropTable(SESSION, first);
+        this.jdbcClient.dropTable(SESSION, second);
+    }
+
+    private CachingJdbcClient cachingStatisticsAwareJdbcClient(Duration duration, boolean cacheMissing)
+    {
+        JdbcClient jdbcClient = database.getJdbcClient();
+        JdbcClient statsAwareJdbcClient = new ForwardingJdbcClient() {
+            @Override
+            protected JdbcClient delegate()
+            {
+                return jdbcClient;
+            }
+
+            @Override
+            public TableStatistics getTableStatistics(ConnectorSession session, JdbcTableHandle handle, TupleDomain<ColumnHandle> tupleDomain)
+            {
+                return NON_EMPTY_STATS;
+            }
+        };
+        return new CachingJdbcClient(statsAwareJdbcClient, Set.of(getTestSessionPropertiesProvider()), duration, cacheMissing);
+    }
+
+    @Test
+    public void testCacheEmptyStatistics()
+    {
+        CachingJdbcClient cachingJdbcClient = createCachingJdbcClient(FOREVER, true);
+        ConnectorSession session = createSession("table");
+
+        JdbcTableHandle table = createTable(new SchemaTableName(schema, "table"));
+
+        int expectedHit = 0, expectedMiss = 0;
```

Comment:
I believe it is discouraged to declare multiple variables in the same line in Java

Reply:
This is perfectly fine here.

Reply:
I don't like it, but I don't want to refactor this test class now. It is used in all the places

Reply:
It's against the code style.

Reply:
https://github.com/airlift/airbase/blob/d32eb78d2381d188883f7ea60ace86d4cdd5beec/airbase-policy/src/main/resources/checkstyle/airbase-checks.xml#L170

Reply:
It's pity.